content/docs/capabilities/mcp.mdx

```diff
@@ -0,0 +1,727 @@
+          scopes: ['read:user', 'user:email']
+          endpoint:
+            auth_url: 'https://github.com/login/oauth/authorize'
+            token_url: 'https://github.com/login/oauth/access_token'
+```
+
+### 3. Building MCP-Enabled Applications
+
+For internal applications that need to call MCP servers through AI APIs, Pomerium provides external tokens that can be passed to LLM services.
+
+```mermaid
+sequenceDiagram
+  actor U as User
+  participant P as Pomerium
+  participant C as Your App Backend
+  participant S as MCP Server
+  participant I as LLM API
+  U ->> P: GET https://mcp-app-demo.your-domain.com
+  P ->> C: GET http://mcp-app-demo:3000 Authorization: Bearer (TE)
+  C ->> I: call tool https://mcp-server.your-domain Authorization: Bearer (TE)
+  I ->> P: GET https://mcp-server.your-domain Authorization: Bearer (TE)
+  P ->> S: GET https://mcp-server
+```
+
+**Route Configuration:**
+
+```yaml
+routes:
+  - from: https://mcp-app-demo.your-domain.com
+    to: http://mcp-app-demo:3000
+    mcp:
+      client: {}
+    policy: {}
+  - from: https://mcp-server.your-domain.com
+    to: http://mcp-server.int:8080/mcp
+    name: My MCP Server
+    mcp:
+      server: {}
+    policy: {}
+```
+
+## Token Types
+
+Understanding Pomerium's token system is crucial for MCP integration:
+
+### External Token (TE)
+
+An externally-facing token issued by Pomerium representing the user's session. External clients use this token to authenticate requests to Pomerium-protected MCP servers.
+
+**Use cases:**
+
+- Providing to LLM APIs for MCP server access
+- Authentication between external MCP clients and Pomerium
+- Internal applications calling external AI services
+
+### Internal Token (TI)
+
+An internal authentication token that Pomerium obtains from upstream OAuth providers on behalf of the user. This token is never exposed to external clients and is used by Pomerium to authenticate with upstream services.
+
+**Characteristics:**
+
+- Automatically refreshed by Pomerium
+- Securely stored and managed
+- Used for upstream API authentication
+- Invisible to external clients
+
+## Configuration Options
+
+### MCP Server Configuration
+
+For routes hosting MCP servers:
+
+```yaml
+mcp:
+  server:
+    # Optional: Configure upstream OAuth2 for services requiring authentication
+    upstream_oauth2:
+      client_id: 'your-oauth-client-id'
+      client_secret: 'your-oauth-client-secret'
+      endpoint:
+        auth_url: 'https://provider.com/oauth/authorize'
+        token_url: 'https://provider.com/oauth/token'
+        auth_style: 'header' # or "params"
+      scopes: ['scope1', 'scope2']
+
+    # Optional: Maximum request body size (default: 1MB)
+    max_request_bytes: 1048576
+```
+
+### MCP Client Configuration
+
+For applications that consume MCP servers:
+
+```yaml
+mcp:
+  # Provides the application with external tokens for MCP server access
+  client: {}
+```
+
+### OAuth2 Endpoint Authentication Styles
```

Comment:
Where is this configurable? This seems out of place without a pomerium YAML configuration snippet example.